search/src/RawImage.cpp

```diff
@@ -194,31 +194,55 @@ void RawImage::applyMask(int flags, const std::vector<int>& exceptions,
 	}
 }
 
-void RawImage::growMask()
+/* This implementation of growMask is optimized for steps > 1
+   (which is how the code is generally used. If you are only
+   growing the mask by 1, the extra copy will be a little slower.
+*/
+void RawImage::growMask(int steps)
 {
-	// Parallel?
-  // This function requires a temporary mask to store the FLAGGED values
-  // Without it, the function will fail to flag all pixels around the masked
-  // regions. Notably, it fails to flag pixels "down" and "right" of the
-  // masked regions.
-  std::vector<float> tmpMask;
-  tmpMask = pixels;
-	for (int i=0; i<width; ++i)
-	{
-		for (int j=0; j<height; j++)
-		{
-			int center = width*j+i;
-			if (i+1<width && pixels[center+1] == NO_DATA) { tmpMask[center] = FLAGGED; continue; }
-			if (i-1>=0 && pixels[center-1] == NO_DATA) { tmpMask[center] = FLAGGED; continue; }
-			if (j+1<height && pixels[center+width] == NO_DATA) { tmpMask[center] = FLAGGED; continue; }
-			if (j-1>=0 && pixels[center-width] == NO_DATA) { tmpMask[center] = FLAGGED; continue; }
-		}
-	}
-  for(std::size_t i=0; i < pixels.size(); ++i)
-  {
-        if (tmpMask[i]==FLAGGED) { pixels[i] = NO_DATA; }
-  }
+    const int num_pixels = width * height;
+    std::vector<int> masked(num_pixels, -1);
+
+    // Set up the initial masked vector that stores the number of steps
+    // each pixel is from a masked pixel in the original image.
+    for (int x = 0; x < width; ++x)
```

Comment:
minor: this could be simplified into a single loop like the bottom one

Reply:
Good call. I made that change.